src/compiler/program.ts

```diff
@@ -1517,8 +1574,28 @@ namespace ts {
             }
         }
 
+        function createRedirectSourceFile(redirectTo: SourceFile, underlying: SourceFile, fileName: string, path: Path): SourceFile {
+            sourceFileIsRedirectedToSet.set(redirectTo.path, true);

```

Comment:
On second thought, it would be better to do all of that *outside* since it makes more sense in the context of `findSourceFile`.